File: 02. C# Part2/01. Arrays-Homework/11. BinarySearch/BinarySearch.cs
using System;

    class BinarySearch
    {
        static void Main()
        {
            //Write a program that finds the index of given element in a sorted array of integers by using the Binary search algorithm.

            int[] numbers = new int[19] { 0, 13, 2, 6, 9, 5, 45, 7, 8, 4, 10, 18, 58, 35, -8, -378, 77, 44, 55 };
            int findNum = 35;
            int index = 0;
            Array.Sort(numbers);
            int center = numbers.Length / 2;
            int currentNum = numbers[center];
            while (findNum != currentNum)
            {
                while (findNum > currentNum)
                {
                    center = (numbers.Length + center) / 2;
                    currentNum = numbers[center];
                }
                while (findNum < currentNum)
                {
                    center = (center) / 2;
                    currentNum = numbers[center];
                }
                index = center;
            }
            Console.WriteLine("{0} is on index {1}", findNum, index);
        }
    }

File: 06. JavaScript Basics/05. Arrays-Homework/05. Arrays-Homework/07. BinarySearch/BinarySearch.js
//Write a script that finds the index of given element in a sorted array of integers by using the binary search algorithm.

function binaryIndexOf(arr, searchElement) {

    var min = 0, max = arr.length - 1, answer = -1;
    while (min <= max) {
        var mid = min + (max - min) / 2;
        mid = Math.floor(mid);
        if (arr[mid] > searchElement) {
            max = mid - 1;
        }
        else if (arr[mid] < searchElement) {
            min = mid + 1;
        }
        else {
            answer = mid;
            break;
        }
    }
    jsConsole.writeLine("From the array: " + arr);
    jsConsole.writeLine("Index of element [" + searchElement + "] is " + answer);
    
}
binaryIndexOf([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], 7);